tests/internal/test_tracer_flare.py

```diff
@@ -93,92 +97,143 @@ def test_single_process_partial_failure(self):
         """
         processes = []
-
-        def do_tracer_flare(prep_request, send_request):
-            self.flare.prepare(prep_request)
-            # Assert that only one process wrote its file successfully
-            # We check for 2 files because it will generate a log file and a config file
-            assert 2 == len(os.listdir(self.flare_dir))
-            self.flare.send(send_request)
+        flares = []
+        for _ in range(2):
+            flares.append(
+                Flare(
+                    trace_agent_url=TRACE_AGENT_URL,
+                    flare_dir=pathlib.Path(self.shared_dir.name),
+                    ddconfig={"config": "testconfig"},
+                )
+            )
+
+        def do_tracer_flare(log_level: str, send_request: FlareSendRequest, flare: Flare):
+            try:
+                flare.prepare(log_level)
+                # Assert that only one process wrote its file successfully
+                # We check for 2 files because it will generate a log file and a config file
+                assert 2 == len(os.listdir(self.shared_dir.name))
+                flare.send(send_request)
+            except Exception as e:
+                self.errors.put(e)
 
         # Create successful process
-        p = multiprocessing.Process(target=do_tracer_flare, args=("DEBUG", self.mock_flare_send_request))
+        p = multiprocessing.Process(target=do_tracer_flare, args=("DEBUG", MOCK_FLARE_SEND_REQUEST, flares[0]))
         processes.append(p)
         p.start()
         # Create failing process
-        p = multiprocessing.Process(target=do_tracer_flare, args=(None, self.mock_flare_send_request))
+        p = multiprocessing.Process(target=do_tracer_flare, args=(None, MOCK_FLARE_SEND_REQUEST, flares[1]))
         processes.append(p)
         p.start()
         for p in processes:
             p.join()
+        assert self.errors.qsize() == 1
 
-    def test_no_app_logs(self):
-        """
-        Validate that app logs are not being added to the
-        file, just the tracer logs
-        """
-        app_logger = Logger(name="my-app", level=DEBUG_LEVEL_INT)
-        self.flare.prepare("DEBUG")
-
-        app_log_line = "this is an app log"
-        app_logger.debug(app_log_line)
 
-        assert os.path.exists(self.flare_file_path)
-
-        with open(self.flare_file_path, "r") as file:
-            for line in file:
-                assert app_log_line not in line, f"File {self.flare_file_path} contains excluded line: {app_log_line}"
+class MockPubSubConnector(PublisherSubscriberConnector):
+    def __init__(self):
+        pass
 
-        self.flare.clean_up_files()
-        self.flare.revert_configs()
+    def read(self):
+        pass
 
-    def confirm_cleanup(self):
-        assert not self.flare.flare_dir.exists(), f"The directory {self.flare.flare_dir} still exists"
-        assert self._get_handler() is None, "File handler was not removed"
+    def write(self):
+        pass
```

Comment:
Had to do this to avoid some issues when trying to instantiate a real `PublisherSubscriberConnector` while using `pyfakefs`